scripts/src/public/services/users.ts

```diff
@@ -92,23 +92,38 @@ export type UpdateUserProps = {
 export async function updateUser(user: User, props: UpdateUserProps) {
 	if (props.walletAddress) {
 		const wallets = await user.getWallets();
-		const app = await Application.get(user.appId);
+		const appId = user.appId;
+		const app = await Application.get(appId);
+		const totalWalletCount = wallets.count + user.walletCount;
+		const walletAddress = props.walletAddress;
+
+		/*  Start of Cross-app restore check */
+		const appWallet = await WalletApplication.findOne({ walletAddress });
+		if (!appWallet) {
+			const newWallet = WalletApplication.create({ walletAddress, appId });
+			await newWallet.save();
+		} else {
+			if (appWallet.appId !== appId) {
+				throw CrossAppWallet();
+			}
+		}
+		/*  End of Cross-app restore check */
```

Comment:
instead of putting comments around this section, you can extract it into a function

Reply:
I would still leave the comments around the function call, just to make it clearer so I don't see any benefit. If you really want I'll do it.

Reply:
i also don't like it too much.
the first comment is ok, but the end part is just weird.

Reply:
It's just to confine the code associated with this so it's clear what needs to be delete when we eventually need to remove it in a few months.

Reply:
And also that development until then won't change/intermingled with this code by mistake.